src/popup/Popup.tsx

```diff
@@ -0,0 +1,202 @@
+                <Notification withCloseButton={false} color="red" title="Error">
+                    Failed to read data from chrome's local storage.
+                </Notification>
+            </div>
+        );
+    }
+
+    return companyNames.length > 0 ? (<CompaniesPanel companyNames={companyNames} />) : (<DownloadCSVPanel setcompanyNames={setcompanyNames} />);
+};
+
+const CompaniesPanel: React.FC<{ companyNames: string[] }> = ({ companyNames }) => {
+    const [searchText, setSearchText] = React.useState("");
+    const filteredCompanyNames = React.useMemo(() => {
+        console.log('searchText', searchText);
+        console.log('companyNames.lengths', companyNames.length);
+        const lowerCaseText = searchText.toLowerCase();
+        if (searchText === "") {
+            return companyNames;
+        }
+        return companyNames.filter(name => name.toLowerCase().includes(lowerCaseText));
+    }, [companyNames, searchText]);
+
+    const tableData: TableData = React.useMemo(() => ({
+        body: filteredCompanyNames.slice(0, 50).map(name => [<Text key={name}>{name}</Text>]),
+    }), [filteredCompanyNames]);
+
+    return (
+        <>
+            <div className="padded">
+                <Text fw={500}>Licensed Companies</Text>
+                <FocusTrap active>
+                    <Input
+                        placeholder="Search company names"
+                        value={searchText}
+                        onChange={e => setSearchText(e.target.value)}
+                        mt="md"
+                    />
+                </FocusTrap>
+                {/* TODO: Pluralize this correctly */}
+                {/* TODO: Format the number nicely */}
+                <Text c="dimmed" size="sm"><b>{filteredCompanyNames.length}</b> companies matching your filter</Text>
+            </div>
+            <Table.ScrollContainer minWidth={200}>
+                <Table highlightOnHover data={tableData} />
+            </Table.ScrollContainer>
+        </>
+    );
+}
+
+const DownloadCSVPanel: React.FC<{setcompanyNames: (companyNames: string[]) => void}> = ( { setcompanyNames }) => {
+    const [url, setUrl] = React.useState("");
+    const { setTrue: setLoading, setFalse: setLoaded, value: isLoading } = useBoolean();
+
+    const handleDownloadClicked = React.useCallback(async () => {
+        setLoading();
+        try {
+            const companyNames = await fetchCsv(url);
+            // const companies = await metchCsv(MOCK_ROWS.join("\n"));
+            await setCompanyData({ companies: companyNames });
+            setLoaded();
+            setcompanyNames(companyNames);
+            console.log("companyNames", companyNames);
+        }
+        catch (e) {
+            console.log("fuck", e);
+            setLoaded();
+        }
+    }, [url]);
+
+    return (
+        <>
+            {/* TODO: Maybe be more helpful here? Render a hyperlink to Uk.gov"s page? Through which the users can grab the csv url? */}
```

Comment:
Remove all TODO items from the code, in favor of creating issues/docs to track these.